client/js/lounge.js

```diff
@@ -646,7 +672,15 @@ $(function() {
 		}
 	});
 
-	socket.on("names", renderChannelUsers);
+	socket.on("names", function(data) {
+		renderChannelUsers(data);
+
+		// Condensed actions support
+		var messages = chat.find("#chan-" + data.id).find(".messages");
+		data.users.forEach(function(user) {
+			messages.find(".user[data-name='" + user.name + "']").removeClass("disconnected");
```

Comment:
As the comment say - it's for future condensed actions. Condensing happens after messages rendering, so colors in that message are not updated. 

To be honest, I don't like this implementation, it might be better to just call `updateDisconnectedUsers(channel, message)` in `setTimeout`